### PR TITLE
Github Leaks OSINT Modules

### PR DESCRIPTION
**Pull request by prince-7**
Module for task mentioned in #106

### CODE REVIEW

File: modules/osint/github_leaks.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+	self.save_gather(output,
+	 'osint/github_leaks', query, output=self.options.get('output'))
+	return output
+
+def module_run(self):
+	output = module_api(self)
+	self.alert("Links\n")
+	for url in output['links']:
+		self.output('\033[1m' + url + '\033[0m', 'G')
+		for line in output['links'][url]:
```
Comment (saeeddhqan) [inline modules/osint/github_leaks.py:136]: Do it in the module_api if the re.sub replace is important.

File: modules/osint/github_leaks.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+	self.save_gather(output,
+	 'osint/github_leaks', query, output=self.options.get('output'))
+	return output
+
+def module_run(self):
+	output = module_api(self)
+	self.alert("Links\n")
+	for url in output['links']:
+		self.output('\033[1m' + url + '\033[0m', 'G')
```
Comment (saeeddhqan) [inline modules/osint/github_leaks.py]: remove colors

File: modules/osint/github_leaks.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+def checks(self, repo):
+	threadpool = concurrent.futures.ThreadPoolExecutor(max_workers=16)
+	futures = (threadpool.submit(leaks, self, link) for link in repo)
+	for results in concurrent.futures.as_completed(futures):
+		print(f"Found {len(OUTPUT['links'])} links" , end= '\r')
+	print('\n')
+
+def leaks(self, repo):
+	checks = ['password','npmrc _auth','dockercfg','pem private','id_rsa','aws_access_key_id','s3cfg','htpasswd','git-credentials','bashrc password','sshd_config','xoxp OR xoxb OR xoxa','SECRET_KEY','client_secret','sshd_config','github_token','api_key','FTP','app_secret','passwd','s3.yml','.exs','beanstalkd.yml','deploy.rake','mysql','credentials','PWD','deploy.rake','.bash_history','.sls','secrets','composer.json']
```
Comment (saeeddhqan) [inline modules/osint/github_leaks.py]: Check the advance search in github. we don't need to search all of them separately!
e.g: if these are keyword: _password OR npmrc _auth OR id_rsa_
if these are filename: `filename:composer.json filename:deploy.rake filename:s3.yml`
if these are extension: `extension:sls extension:exs`
Categorize them based on the filename, extension, keyword.
deploy.rake has been duplicated.

File: modules/osint/github_leaks.py (hunk 1 above)
Comment (prince-7) [reply modules/osint/github_leaks.py:136]: done

File: modules/osint/github_leaks.py (hunk 2 above)
Comment (prince-7) [reply modules/osint/github_leaks.py]: done

File: modules/osint/github_leaks.py (hunk 3 above)
Comment (prince-7) [reply modules/osint/github_leaks.py]: this can't be done for keywords, rest done

File: modules/osint/github_leaks.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+OUTPUT = {'links': {}}
+
+def checks(self, repo):
+	checks = ['filename:.npmrc filename:.dockercfg filename:id_rsa filename:credentials filename:.s3cfg',#NOT MORE THAN 500 CHAR IN ONE QUERY
+	'filename:wp-config.php filename:.htpasswd filename:.git-credentials filename:.bashrc filename:.bash_profile', 
+	'filename:.netrc filename:config filename:connections.xml filename:express.conf filename:.pgpass', 
+	'filename:proftpdpasswd filename:server.cfg filename:.bash_history filename:.cshrc filename:.history filename:.sh_history filename:sshd_config', 
+	'filename:dhcpd.conf filename:prod.exs filename:shadow filename:passwd filename:.esmtprc filename:logins.json', 
+	'filename:CCCam.cfg filename:settings filename:secrets filename:master.key filename:WebServers.xml',
+	'password', 'DB_USERNAME', 'TOKEN', 'API_KEY' , 'SECRET_KEY', 'credentials',
```
Comment (saeeddhqan) [inline modules/osint/github_leaks.py:48]: Doesn't it work? `password OR DB_USERNAME OR TOKEN...`

Comment (prince-7) [reply modules/osint/github_leaks.py:48]: No. for eg. https://github.com/saeeddhqan/Maryam/search?q=page shows results but https://github.com/saeeddhqan/Maryam/search?q=page+OR+somethingthatdoesn%27texist does not show results.

**Pull request merged**